Latest version 2.18.0 (master branch) is getting panic

### Summary 
We are getting a panic issue on 2.18.0 version. 2.17.0 is working fine.

```
[gosec] 2023/10/10 04:57:21 Including rules: default
[gosec] 2023/10/10 04:57:21 Excluding rules: default
[gosec] 2023/[10](https://github.com/***/****/actions/runs/*******/job/*******?pr=1649#step:6:11)/10 
[gosec] 2023/10/10 04:57:35 Import directory: /github/workspace/internal/temporal/activity
panic: interface conversion: types.Type is *types.Basic, not *types.Signature
```


### Steps to reproduce the behavior

### gosec version
2.18.0

### Go version (output of 'go version')
go1.21.1

### Operating system / Environment
ubuntu22-6

### Expected behavior

### Actual behavior

we are also getting panic.

`go1.21.1 linux/amd64` and `securego/gosec@master` running from a GitHub action 
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x62c55d]

goroutine 1 [running]:
golang.org/x/tools/go/ssa.(*Program).packageLevelMember(...)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/source.go:186
golang.org/x/tools/go/ssa.(*builder).expr0(0xc000bb7648, 0xc0006a5b00, {0x8e54b0?, 0xc0000d1320}, {0x0, {0x0, 0x0}, {0x0, 0x0}})
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:796 +0x123d
golang.org/x/tools/go/ssa.(*builder).expr(0x7ccac0?, 0xc0006a5b00, {0x8e54b0, 0xc0000d1320})
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:623 +0x175
golang.org/x/tools/go/ssa.(*builder).expr0(0xc000bb7648, 0xc0006a5b00, {0x8e5540?, 0xc000416cc0}, {0x0, {0x0, 0x0}, {0x0, 0x0}})
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:818 +0x18d1
golang.org/x/tools/go/ssa.(*builder).expr(0xc000bb6b88?, 0xc0006a5b00, {0x8e5540, 0xc000416cc0})
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:623 +0x175
golang.org/x/tools/go/ssa.(*builder).addr(0xc000bb6c68?, 0xc0006a5b00, {0x8e56f0?, 0xc000416cd8}, 0x0)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:512 +0xb0b
golang.org/x/tools/go/ssa.(*builder).expr0(0xc000bb7648, 0xc0006a5b00, {0x8e56f0?, 0xc000416cd8}, {0x0, {0x0, 0x0}, {0x0, 0x0}})
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:935 +0x1e0d
golang.org/x/tools/go/ssa.(*builder).expr(0xc000476c30?, 0xc0006a5b00, {0x8e56f0, 0xc000416cd8})
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:623 +0x175
golang.org/x/tools/go/ssa.(*builder).setCallFunc(0x7f277656d200?, 0x7f27bd125108?, 0x80?, 0xc0006f9fc0)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:1039 +0x314
golang.org/x/tools/go/ssa.(*builder).setCall(0x7cdd20?, 0xc000305a70?, 0xc0001e5ec0, 0xc0006f9fc0)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:1118 +0x2e
golang.org/x/tools/go/ssa.(*builder).expr0(0xc000bb7648, 0xc0006a5b00, {0x8e5570?, 0xc0001e5ec0}, {0x0, {0x0, 0x0}, {0x0, 0x0}})
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:702 +0x249e
golang.org/x/tools/go/ssa.(*builder).expr(0x40d1da?, 0xc0006a5b00, {0x8e5570, 0xc0001e5ec0})
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:623 +0x175
golang.org/x/tools/go/ssa.(*builder).assign(0xc0006e9600?, 0x8e7870?, {0x8e6660?, 0xbc2fa0}, {0x8e5570?, 0xc0001e5ec0?}, 0x80?, 0x0)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:596 +0x3a5
golang.org/x/tools/go/ssa.(*Package).build(0xc000481a80)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:2532 +0xb0d
sync.(*Once).doSlow(0xc0001320c0?, 0xc00041d8c0?)
	/usr/local/go/src/sync/once.go:74 +0xbf
sync.(*Once).Do(...)
	/usr/local/go/src/sync/once.go:65
golang.org/x/tools/go/ssa.(*Package).Build(...)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:2455
golang.org/x/tools/go/analysis/passes/buildssa.run(0xc000692000)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/analysis/passes/buildssa/buildssa.go:71 +0x18d
github.com/securego/gosec/v2.(*Analyzer).CheckAnalyzers(0xc00016be80, 0xc0001c4a80)
	/build/analyzer.go:341 +0x1b0
github.com/securego/gosec/v2.(*Analyzer).Process(0xc00016be80, {0x0, 0x0, 0x0}, {0xc000025800, 0x3a, 0x3d?})
	/build/analyzer.go:234 +0x478
main.main()
	/build/cmd/gosec/main.go:425 +0xc3c
```

This is the same failure but on different file. 
```
[gosec] 2023/10/10 04:20:27 Including rules: default
[gosec] 2023/10/10 04:20:27 Excluding rules: default
[gosec] 2023/10/10 04:20:51 Checking file: /github/workspace/pkg/cloudevents/options.go
panic: interface conversion: types.Type is *types.Basic, not *types.Signature

goroutine 1 [running]:
golang.org/x/tools/go/ssa.(*builder).expr0(0xc002425648, 0xc000cbb200, {0x8e54b0?, 0xc001bbd3e0}, {0x2, {0x8e4fb8, 0xb856e0}, {0x0, 0x0}})
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:791 +0x2266
golang.org/x/tools/go/ssa.(*builder).expr(0x629170?, 0xc000cbb200, {0x8e54b0, 0xc001bbd3e0})
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:623 +0x175
golang.org/x/tools/go/ssa.(*builder).setCallFunc(0x7f091bd66b78?, 0x7f0962b39108?, 0x80?, 0xc0037024c0)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:1039 +0x314
golang.org/x/tools/go/ssa.(*builder).setCall(0x7ccac0?, 0xc000cbb200?, 0xc000c171c0, 0xc0037024c0)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:[11](https://github.com/anzx/lex-settlement-services/actions/runs/6464669446/job/17549661650?pr=1649#step:6:12)18 +0x2e
golang.org/x/tools/go/ssa.(*builder).expr0(0xc002425648, 0xc000cbb200, {0x8e5570?, 0xc000c171c0}, {0x0, {0x0, 0x0}, {0x0, 0x0}})
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:702 +0x249e
golang.org/x/tools/go/ssa.(*builder).expr(0xc002424f10?, 0xc000cbb200, {0x8e5570, 0xc000c171c0})
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:623 +0x175
golang.org/x/tools/go/ssa.(*builder).assign(0xc002425020?, 0xc000cbb200?, {0x8e65f0?, 0xc0019fa600}, {0x8e5570?, 0xc000c171c0?}, 0x0?, 0xc002425008)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:596 +0x3a5
golang.org/x/tools/go/ssa.(*builder).assignStmt(0xc000cbb200?, 0xc000cbb200, {0xc0006eea10, 0x1, 0x816580?}, {0xc0006eea40, 0x1, 0x417ee8?}, 0x0)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:[12](https://github.com/anzx/lex-settlement-services/actions/runs/6464669446/job/17549661650?pr=1649#step:6:13)03 +0x3b8
golang.org/x/tools/go/ssa.(*builder).stmt(0x1?, 0xc000cbb200, {0x8e5b40?, 0xc000c17200?})
	/go/pkg/mod/golang.org/x/tools@v0.[14](https://github.com/anzx/lex-settlement-services/actions/runs/6464669446/job/17549661650?pr=1649#step:6:15).0/go/ssa/builder.go:2[15](https://github.com/anzx/lex-settlement-services/actions/runs/6464669446/job/17549661650?pr=1649#step:6:16)9 +0x42d
golang.org/x/tools/go/ssa.(*builder).stmtList(...)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:944
golang.org/x/tools/go/ssa.(*builder).stmt(0xc000cbb200?, 0xc000cbb200, {0x8e5c60?, 0xc0019cabd0?})
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:2255 +0xe30
golang.org/x/tools/go/ssa.(*builder).buildFunctionBody(0x644ece?, 0xc000cbb200)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:2369 +0x410
golang.org/x/tools/go/ssa.(*builder).buildFunction(0x644f20?, 0xc000cbb200)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:2304 +0x28
golang.org/x/tools/go/ssa.(*builder).buildCreated(0xc002425648)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:2391 +0x25
golang.org/x/tools/go/ssa.(*Package).build(0xc0002c5b80)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:2584 +0xc58
sync.(*Once).doSlow(0xc001a583c0?, 0xc000ea9920?)
	/usr/local/go/src/sync/once.go:74 +0xbf
sync.(*Once).Do(...)
	/usr/local/go/src/sync/once.go:65
golang.org/x/tools/go/ssa.(*Package).Build(...)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/ssa/builder.go:2455
golang.org/x/tools/go/analysis/passes/buildssa.run(0xc00007f450)
	/go/pkg/mod/golang.org/x/tools@v0.14.0/go/analysis/passes/buildssa/buildssa.go:71 +0x[18](https://github.com/anzx/lex-settlement-services/actions/runs/6464669446/job/17549661650?pr=1649#step:6:19)d
github.com/securego/gosec/v2.(*Analyzer).CheckAnalyzers(0xc0002c5f00, 0xc003761800)
	/build/analyzer.go:341 +0x1b0
github.com/securego/gosec/v2.(*Analyzer).Process(0xc0002c5f00, {0x0, 0x0, 0x0}, {0xc00057d000, 0xe6, 0x3d?})
	/build/analyzer.go:[23](https://github.com/anzx/lex-settlement-services/actions/runs/6464669446/job/17549661650?pr=1649#step:6:24)4 +0x478
main.main()
	/build/cmd/gosec/main.go:4[25](https://github.com/anzx/lex-settlement-services/actions/runs/6464669446/job/17549661650?pr=1649#step:6:26) +0xc3c
```

Sample Code:

```
package cloudevents

import "errors"

// Options holds static configuration values
type Options struct {
	TopicId        string
	SubscriptionId string
	ProjectId      string
}

// DefaultOptions returns an Options pointer with default values
func DefaultOptions() *Options {
	return &Options{
		TopicId:        "lex",
		ProjectId:      "lex",
		SubscriptionId: "lex",
	}
}

func (opts *Options) Validate() error {
	if opts == nil {
		return errors.New("options cannot be a nil pointer")
	}
	if opts.TopicId == "" && opts.SubscriptionId == "" {
		return errors.New("TopicId and SubscriptionId cannot both be empty for cloudevents")
	}
	if opts.ProjectId == "" {
		return errors.New("ProjectId cannot be empty for cloudevents")
	}
	return nil
}
```

--- a/analyzer.go
+++ b/analyzer.go
@@ -282,7 +282,7 @@ func (gosec *Analyzer) load(pkgPath string, conf *packages.Config) ([]*packages.
 	return pkgs, nil
 }
 
-// CheckRules runs analysis on the given package
+// CheckRules runs analysis on the given package.
 func (gosec *Analyzer) CheckRules(pkg *packages.Package) {
 	gosec.logger.Println("Checking package:", pkg.Name)
 	for _, file := range pkg.Syntax {
@@ -318,31 +318,14 @@ func (gosec *Analyzer) CheckRules(pkg *packages.Package) {
 	}
 }
 
-// CheckAnalyzers runs analyzers on a given package
+// CheckAnalyzers runs analyzers on a given package.
 func (gosec *Analyzer) CheckAnalyzers(pkg *packages.Package) {
-	ssaPass := &analysis.Pass{
-		Analyzer:          buildssa.Analyzer,
-		Fset:              pkg.Fset,
-		Files:             pkg.Syntax,
-		OtherFiles:        pkg.OtherFiles,
-		IgnoredFiles:      pkg.IgnoredFiles,
-		Pkg:               pkg.Types,
-		TypesInfo:         pkg.TypesInfo,
-		TypesSizes:        pkg.TypesSizes,
-		ResultOf:          nil,
-		Report:            nil,
-		ImportObjectFact:  nil,
-		ExportObjectFact:  nil,
-		ImportPackageFact: nil,
-		ExportPackageFact: nil,
-		AllObjectFacts:    nil,
-		AllPackageFacts:   nil,
-	}
-	ssaResult, err := ssaPass.Analyzer.Run(ssaPass)
+	ssaResult, err := gosec.buildSSA(pkg)
 	if err != nil {
-		gosec.logger.Printf("Error running SSA analyser on package %q: %s", pkg.Name, err)
+		gosec.logger.Printf("Error building the SSA representation of the package %q: %s", pkg.Name, err)
 		return
 	}
+
 	resultMap := map[*analysis.Analyzer]interface{}{
 		buildssa.Analyzer: &analyzers.SSAAnalyzerResult{
 			Config: gosec.Config(),
@@ -382,6 +365,35 @@ func (gosec *Analyzer) CheckAnalyzers(pkg *packages.Package) {
 			}
 		}
 	}
+}
+
+// buildSSA runs the SSA pass which builds the SSA representation of the package. It handles gracefully any panic.
+func (gosec *Analyzer) buildSSA(pkg *packages.Package) (interface{}, error) {
+	defer func() {
+		if r := recover(); r != nil {
+			gosec.logger.Printf("Panic when running SSA analyser on package %q: %s", pkg.Name, r)
+		}
+	}()
+	ssaPass := &analysis.Pass{
+		Analyzer:          buildssa.Analyzer,
+		Fset:              pkg.Fset,
+		Files:             pkg.Syntax,
+		OtherFiles:        pkg.OtherFiles,
+		IgnoredFiles:      pkg.IgnoredFiles,
+		Pkg:               pkg.Types,
+		TypesInfo:         pkg.TypesInfo,
+		TypesSizes:        pkg.TypesSizes,
+		ResultOf:          nil,
+		Report:            nil,
+		ImportObjectFact:  nil,
+		ExportObjectFact:  nil,
+		ImportPackageFact: nil,
+		ExportPackageFact: nil,
+		AllObjectFacts:    nil,
+		AllPackageFacts:   nil,
+	}
+
+	return ssaPass.Analyzer.Run(ssaPass)
 }
 
 func isGeneratedFile(file *ast.File) bool {

